specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2018-01-01-preview/schemaregistry.json

```diff
@@ -1,21 +1,21 @@
 {
   "swagger": "2.0",
   "info": {
+    "title": "Azure Schema Registry",
     "version": "2018-01-01-preview",
```

Comment:
The current implementation wants version 2017-04, will it be updated to accept 2018-01-01-preview soon?

Unfortunately, passing apiVersion to autogenerated client doesn't work, I think it may be a bug in the TS v6 generator, but I'm not sure. Also, transforming the swagger to have 2017-04 causes autorest to crash. I'm currently just patching the autogenerated code to unblock myself.

Reply:
@hmlam can we support 2018-01-01-preview as service version?

Reply:
We really should create new service version so we can do this swagger the right way.  I'm just not seeing leaving 2017-04 as a good option any longer.

Reply:
@nguerrera AutoRest crashes because 2017-04 isn't a semantic version. However, your generated code should have `apiVersion` as a parameter to your client, since it is defined that way in the swagger.

Reply:
@MiYanni. It does, but passing 2017-04 does nothing AFAICT, and it still sends the defaultValue in the parameter spec. I think there may be a bug in the generator as I said above. I will try to isolate a repro and file a bug on the generator if I can confirm the issue.

Reply:
(Aside: Dare I ask why we're using 2018-* for something that's only previewing now?)

Reply:
That's a good question honestly... I just assumed that was the convention.  We should do 20-09-01-preview.

Reply:
So for now, we should be able to continue using the existing API with x-schema-type and 2017-04.

Once 2020-09-01-preview is in prod, we will change the swagger and we can generate a new version of the API.

I'll leave this comment open as a reminder.